application/temperature.py

```diff
@@ -0,0 +1,26 @@
+import httpx
+from sqlalchemy.orm import Session
+from datetime import datetime
+from application import models, schemas
+
+
+async def fetch_temperature(city_name: str):
+    response = await httpx.get(f"http://api.openweathermap.org/data/2.5/weather?q={city_name}&appid=YOUR_API_KEY")
```

Comment:
The API key is hardcoded in the URL. It's a security risk to expose API keys in the source code. Consider using environment variables or a configuration file to manage sensitive information.